src/Governance.ts

```diff
@@ -89,7 +103,44 @@ export class Governance {
       })
     );
 
-    return result.data.proposals.map(({ pipId }) => new Proposal({ pipId }, context));
+    const proposalsWithDetails = result.data.proposals.map(
+      ({
+        pipId,
+        proposer: proposerAddress,
+        createdAt,
+        url: discussionUrl,
+        description,
+        coolOffEndBlock,
+        endBlock,
+        proposal: rawProposal,
+        lastState,
+        lastStateUpdatedAt,
+        totalVotes,
+        totalAyesWeight,
+        totalNaysWeight,
+      }) => {
+        const proposal = new Proposal({ pipId }, context);
+        return {
+          proposal,
+          details: {
+            proposerAddress,
+            createdAt: new BigNumber(createdAt),
+            discussionUrl,
+            description,
+            coolOffEndBlock: new BigNumber(coolOffEndBlock),
+            endBlock: new BigNumber(endBlock),
+            transaction: rawProposal ? middlewareProposalToTxTag(rawProposal, context) : null,
+            lastState,
+            lastStateUpdatedAt: new BigNumber(lastStateUpdatedAt),
+            totalVotes: new BigNumber(totalVotes),
+            totalAyesWeight: new BigNumber(totalAyesWeight),
+            totalNaysWeight: new BigNumber(totalNaysWeight),
+          },
+        };
```

Comment:
This code is too similar to the one in `getDetails`. You should create a util, something like `middlewareProposalToProposalDetails` and use it in both places